packages/connect-node/src/grpc-http-transport.ts

```diff
@@ -191,12 +232,20 @@ export function createGrpcHttpTransport(
               throw "premature eof";
             }
 
+            let responseBody = messageResult.value.data;
+            if (compression) {
+              responseBody = await compression.decompress(
+                responseBody,
+                readMaxBytes
+              );
+            }
```

Comment:
You have to check the envelope flags. You must only decompress if the compressed flag is set.

I think it's a misnomer to use `responseBody` here. (Connect unary is different because it is not enveloped). Can you rename to `messageData`?

You can change the order a bit to make this more readable:

```ts
            const messageResult = await readEnvelope(response);
            if (messageResult.done) {
              throw "premature eof";
            }
            let messageData = messageResult.value.data;
            if (compressed flag is set) {
              throw error if grpc-encoding not set
              messageData = decompress(messageData)
            }

            const eofResult = await readEnvelope(response);
            if (!eofResult.done) {
              throw "extraneous data";
            }

            const trailer = await trailerPromise;
            grpcValidateTrailer(trailer);
```

Reply:
hey @timostamm  i tried to follow the ordering you wrote above, however, when we move the `messageResult.done` check below reading of the envelope we fail a lot of test. Would you be ok if i went with this approach?

```ts
            const messageResult = await readEnvelope(response);
            const eofResult = await readEnvelope(response);
            if (!eofResult.done) {
              throw "extraneous data";
            }

            const trailer = await trailerPromise;
            grpcValidateTrailer(trailer);

            if (messageResult.done) {
              throw "premature eof";
            }
            let messageData = messageResult.value.data;
            if (
              (messageResult.value.flags & compressedFlag) ===
              compressedFlag
            ) {
              if (!compression) {
                throw new ConnectError(
                  `received compressed envelope, but no grpc-encoding`,
                  Code.InvalidArgument
                );
              }
              messageData = await compression.decompress(
                messageData,
                readMaxBytes
              );
            }
```

Reply:
Ah, of course we need to check for an error in the trailer first. 
It's not ideal, but yes, let's go with the order you got for now.